### PR TITLE
Add post-increment and reg-reg load/store instructions, and GAS tests

### PR DESCRIPTION
**Pull request by EnricoTabanelli**
gas/ChangeLog.COREV:

	* config/tc-riscv.c (riscv_multi_subset_supports): Add 
	post-increment and register-indexed load/store instruction 
	class. 
	(validate_riscv_insn, riscv_ip): Add post-increment symbol.
	* doc/c-riscv.texi: Added details on CORE-V post-incrementing
	and reg-reg load/store ops ISA options.

gas/testsuite/ChangeLog.COREV:

	* gas/riscv/cv-mem-fail-operand-01.d: Add post-increment
	and register-indexed load/store test.
	* gas/riscv/cv-mem-fail-operand-01.s: Likewise.
	* gas/riscv/cv-mem-fail-operand-01.l: Likewise.
	* gas/riscv/cv-mem-fail-operand-02.d: Likewise.
	* gas/riscv/cv-mem-fail-operand-02.s: Likewise.
	* gas/riscv/cv-mem-fail-operand-02.l: Likewise.
	* gas/riscv/cv-mem-fail-operand-03.d: Likewise.
	* gas/riscv/cv-mem-fail-operand-03.s: Likewise.
	* gas/riscv/cv-mem-fail-operand-03.l: Likewise.
	* gas/riscv/cv-mem-fail-operand-04.d: Likewise.
	* gas/riscv/cv-mem-fail-operand-04.s: Likewise.
	* gas/riscv/cv-mem-fail-operand-04.l: Likewise.
	* gas/riscv/cv-mem-fail-operand-05.d: Likewise.
	* gas/riscv/cv-mem-fail-operand-05.s: Likewise.
	* gas/riscv/cv-mem-fail-operand-05.l: Likewise.
	* gas/riscv/cv-mem-fail-march.d: Likewise.
	* gas/riscv/cv-mem-fail-march.s: Likewise.
	* gas/riscv/cv-mem-fail-march.l: Likewise.
	* gas/riscv/cv-mem-lb.d: Likewise.
	* gas/riscv/cv-mem-lbpost.d: Likewise.
	* gas/riscv/cv-mem-lbpost.s: Likewise.
	* gas/riscv/cv-mem-lbrr.d: Likewise.
	* gas/riscv/cv-mem-lbrrpost.d: Likewise.
	* gas/riscv/cv-mem-lbrrpost.s: Likewise.
	* gas/riscv/cv-mem-lbrr.s: Likewise.
	* gas/riscv/cv-mem-lb.s: Likewise.
	* gas/riscv/cv-mem-lbu.d: Likewise.
	* gas/riscv/cv-mem-lbupost.d: Likewise.
	* gas/riscv/cv-mem-lbupost.s: Likewise.
	* gas/riscv/cv-mem-lburr.d: Likewise.
	* gas/riscv/cv-mem-lburrpost.d: Likewise.
	* gas/riscv/cv-mem-lburrpost.s: Likewise.
	* gas/riscv/cv-mem-lburr.s: Likewise.
	* gas/riscv/cv-mem-lbu.s: Likewise.
	* gas/riscv/cv-mem-lh.d: Likewise.
	* gas/riscv/cv-mem-lhpost.d: Likewise.
	* gas/riscv/cv-mem-lhpost.s: Likewise.
	* gas/riscv/cv-mem-lhrr.d: Likewise.
	* gas/riscv/cv-mem-lhrrpost.d: Likewise.
	* gas/riscv/cv-mem-lhrrpost.s: Likewise.
	* gas/riscv/cv-mem-lhrr.s: Likewise.
	* gas/riscv/cv-mem-lh.s: Likewise.
	* gas/riscv/cv-mem-lhu.d: Likewise.
	* gas/riscv/cv-mem-lhupost.d: Likewise.
	* gas/riscv/cv-mem-lhupost.s: Likewise.
	* gas/riscv/cv-mem-lhurr.d: Likewise.
	* gas/riscv/cv-mem-lhurrpost.d: Likewise.
	* gas/riscv/cv-mem-lhurrpost.s: Likewise.
	* gas/riscv/cv-mem-lhurr.s: Likewise.
	* gas/riscv/cv-mem-lhu.s: Likewise.
	* gas/riscv/cv-mem-lw.d: Likewise.
	* gas/riscv/cv-mem-lwpost.d: Likewise.
	* gas/riscv/cv-mem-lwpost.s: Likewise.
	* gas/riscv/cv-mem-lwrr.d: Likewise.
	* gas/riscv/cv-mem-lwrrpost.d: Likewise.
	* gas/riscv/cv-mem-lwrrpost.s: Likewise.
	* gas/riscv/cv-mem-lwrr.s: Likewise.
	* gas/riscv/cv-mem-lw.s: Likewise.
	* gas/riscv/cv-mem-march-rv32i-xcorev.d: Likewise.
	* gas/riscv/cv-mem-march-rv32i-xcorev.s: Likewise.
	* gas/riscv/cv-mem-sb.d: Likewise.
	* gas/riscv/cv-mem-sbpost.d: Likewise.
	* gas/riscv/cv-mem-sbpost.s: Likewise.
	* gas/riscv/cv-mem-sbrr.d: Likewise.
	* gas/riscv/cv-mem-sbrrpost.d: Likewise.
	* gas/riscv/cv-mem-sbrrpost.s: Likewise.
	* gas/riscv/cv-mem-sbrr.s: Likewise.
	* gas/riscv/cv-mem-sb.s: Likewise.
	* gas/riscv/cv-mem-sh.d: Likewise.
	* gas/riscv/cv-mem-shpost.d: Likewise.
	* gas/riscv/cv-mem-shpost.s: Likewise.
	* gas/riscv/cv-mem-shrr.d: Likewise.
	* gas/riscv/cv-mem-shrrpost.d: Likewise.
	* gas/riscv/cv-mem-shrrpost.s: Likewise.
	* gas/riscv/cv-mem-shrr.s: Likewise.
	* gas/riscv/cv-mem-sh.s: Likewise.
	* gas/riscv/cv-mem-sw.d: Likewise.
	* gas/riscv/cv-mem-swpost.d: Likewise.
	* gas/riscv/cv-mem-swpost.s: Likewise.
	* gas/riscv/cv-mem-swrr.d: Likewise.
	* gas/riscv/cv-mem-swrrpost.d: Likewise.
	* gas/riscv/cv-mem-swrrpost.s: Likewise.
	* gas/riscv/cv-mem-swrr.s: Likewise.
	* gas/riscv/cv-mem-sw.s: Likewise.

include/ChangeLog.COREV:

	* riscv-opc.h: Add post-increment and register-indexed 
	load/store matches and masks.
	* riscv.h (riscv_insn_class): Add post-increment and
	register-indexed load/store class.

opcodes/ChangeLog.COREV:

	* riscv-dis.c (print_insn_args): Add post-increment symbol. 
	* riscv-opc.c (riscv_opcodes): Add post-increment and 
	register-indexed load/store opcodes.

Signed-off-by: Enrico Tabanelli <enrico.tabanelli@embecosm.com>

### CODE REVIEW

File: opcodes/riscv-opc.c [hunk 1]
```diff
@@ -901,6 +901,40 @@ const struct riscv_opcode riscv_opcodes[] =
...
+{"cv.sb", 0, INSN_CLASS_COREV_MEM, "t,d(s!)", MATCH_CV_SBRRPOST,  MASK_CV_SRRPOST,  match_opcode, 0},
+{"cv.sh", 0, INSN_CLASS_COREV_MEM, "t,q(s)",  MATCH_SH,           MASK_SH,       match_opcode, INSN_DREF|INSN_2_BYTE},
+{"cv.sh", 0, INSN_CLASS_COREV_MEM, "t,q(s!)", MATCH_CV_SHPOST,    MASK_CV_SPOST,    match_opcode, 0},
+{"cv.sh", 0, INSN_CLASS_COREV_MEM, "t,d(s)",  MATCH_CV_SHRR,      MASK_CV_SRR,      match_opcode, 0},
+{"cv.sh", 0, INSN_CLASS_COREV_MEM, "t,d(s!)", MATCH_CV_SHRRPOST,  MASK_CV_SRRPOST,  match_opcode, 0},
+{"cv.sw", 0, INSN_CLASS_COREV_MEM, "t,q(s)",  MATCH_SW,           MASK_SW,       match_opcode, INSN_DREF|INSN_4_BYTE},
+{"cv.sw", 0, INSN_CLASS_COREV_MEM, "t,q(s!)", MATCH_CV_SWPOST,    MASK_CV_SPOST,    match_opcode, 0},
+{"cv.sw", 0, INSN_CLASS_COREV_MEM, "t,d(s)",  MATCH_CV_SWRR,      MASK_CV_SRR,      match_opcode, 0},
+{"cv.sw", 0, INSN_CLASS_COREV_MEM, "t,d(s!)", MATCH_CV_SWRRPOST,  MASK_CV_SRRPOST,  match_opcode, 0},
+
```
Comment (jessicamills) [inline opcodes/riscv-opc.c]: The indentation is a little off here

File: include/opcode/riscv-opc.h [hunk 2]
```diff
@@ -958,6 +958,41 @@
...
+#define MATCH_CV_SHRRPOST 0x0000502b
+#define MATCH_CV_SWRRPOST 0x0000602b
+#define MATCH_CV_SBRR     0x00004023
+#define MATCH_CV_SHRR     0x00005023
+#define MATCH_CV_SWRR     0x00006023
+
+#define MASK_CV_SPOST     0x707f
+#define MASK_CV_SRRPOST   0xfe00707f
+#define MASK_CV_SRR       0xfe00707f
+
```
Comment (jessicamills) [inline include/opcode/riscv-opc.h]: There should be consistency with whether we fill with zeros for the whole 32-bits here

Comment (jessicamills) [reply include/opcode/riscv-opc.h]: Looking at MATCHES and MASKS above they seem to not fill with zeros, so lets remove these (I have done the same for the rest of CORE-V https://github.com/jessicamills/corev-binutils-gdb/tree/jessica-bug-fix)

**Review by jessicamills — COMMENTED** [review]
Looks good, just a few formatting comments.

File: opcodes/riscv-opc.c (hunk 1 above)
Comment (EnricoTabanelli) [reply opcodes/riscv-opc.c]: Done

File: include/opcode/riscv-opc.h (hunk 2 above)
Comment (EnricoTabanelli) [reply include/opcode/riscv-opc.h]: Done

**Comment by jessicamills** [comment]
Good to merge. Built branch locally and ran tests:
```
		=== gas Summary ===

# of expected passes		482
# of expected failures		21
# of unsupported tests		5

		=== ld Summary ===

# of expected passes		406
# of expected failures		4
# of untested testcases		26
# of unsupported tests		182
```

**Pull request merged**